GGRC-1294 Filter on `"custom attribute" is empty` should work

dyezepchik opened the pull request:
Precondition:

created GCAs with all the types for e.g. Assessment, created program, control, audit, created assessment template on the audit page with LCAs for all

the types

Steps to reproduce:

1. Generate at least 2 assessments on the audit page based on control and assessment template

2. First assessment: GCAs and LCAs fields are empty in Info pane

3. Second assessment: fill in the values in GCAs and LCAs fields in Info pane

4. Filter on custom attribute = empty

Expected Result: Filter on custom attribute = empty should work

PLEASE NOTE THE FOLLOWING ­

Filtering on attribute = "" is the most important for assessment filter. However due to object level custom attributes, some assessment may not even

have the attribute that the filter is set on. The empty filter MUST ALSO return those assessments that do not even have the attribute.

zidarsk8 commented:
@dyezepchik can you also remove the evaluate function from here:
https://github.com/google/ggrc-core/blob/develop/src/parser/parser_template.js

dyezepchik commented:
Could not add filter test for CAs. Still needs work.

alieh-rymasheuski commented on src/ggrc/converters/query_helper.py:
```diff
@@ -914,10 +913,28 @@ def like(left, right):
+          Record.property == left,
+          sa.not_(sa.or_(Record.content == u"", Record.content.is_(None))),
+      )
+      return object_class.id.in_(db.session.query(Record.key).filter(
```
Please replace this clause with its equivalent `object_class.id.notin_(subquery)`.

zidarsk8 commented on src/ggrc/assets/js_specs/generated/ggrc_filter_query_parser_spec.js:
```diff
@@ -227,120 +208,20 @@ describe('GGRC.query_parser', function() {
-      });
-
-      it("evaluates expressions that end with a full text search", function(){
-        expect(GGRC.query_parser.parse('hello=worldoo or ~ bacon ipsum')
```
I do not think we shoud be removing these tests. This was a simple way of checking if the parsed query was correct, but now we don't have that check anymore. 

Instead of removing them, they should be rewritten to just check the parsed expression.
so removing the `.evaluate` part and changing the `toEqual` part.

alieh-rymasheuski replied:
Then let's move `evaluate` removal out of this PR into a separate one that will include all the required tests.

dyezepchik commented:
I've modified only 2 js files in this PR. One of them by hand and I fixed all the lintings in it within my changes, and even a few more. But the other one is auto generated with ./generate_parser.js. And I suppose code-style fails because of it. And I can not fix "758 problems (570 errors, 188 warnings)" found in it.

alieh-rymasheuski commented on src/ggrc/converters/query_helper.py:
```diff
@@ -914,10 +913,26 @@ def like(left, right):
+            right))
+      subquery = db.session.query(Record.key).filter(
+          Record.type == object_class.__name__,
+          Record.property == left,
```
Please add logic to convert the value of `left` from an alias (like `"Network Zone"`) to field name (like `"network_zone"`).
